apollo-compiler/src/main/kotlin/com/apollographql/apollo/compiler/parser/graphql/ast/document.kt

```diff
@@ -0,0 +1,128 @@
+package com.apollographql.apollo.compiler.parser.graphql.ast
+
+import com.apollographql.apollo.compiler.parser.antlr.GraphQLLexer
+import com.apollographql.apollo.compiler.parser.antlr.GraphQLParser
+import com.apollographql.apollo.compiler.parser.error.DocumentParseException
+import com.apollographql.apollo.compiler.parser.error.ParseException
+import okio.Buffer
+import okio.BufferedSink
+import okio.buffer
+import okio.sink
+import org.antlr.v4.runtime.BaseErrorListener
+import org.antlr.v4.runtime.CharStreams
+import org.antlr.v4.runtime.CommonTokenStream
+import org.antlr.v4.runtime.RecognitionException
+import org.antlr.v4.runtime.Recognizer
+import org.antlr.v4.runtime.Token
+import org.antlr.v4.runtime.atn.PredictionMode
+import java.io.File
+import java.io.InputStream
+import java.io.OutputStream
+
+fun GQLDocument.withBuiltinTypes(): GQLDocument {
+  val buildInsInputStream = javaClass.getResourceAsStream("/builtins.sdl")
+  return copy(
+      definitions = definitions + GQLDocument.parseInternal(buildInsInputStream).definitions
+  )
+}
+
+fun GQLDocument.withoutBuiltinTypes(): GQLDocument {
+  return copy(
+      definitions = definitions.filter {
+        when ((it as? GQLNamed)?.name) {
+          "Int", "Float", "String", "ID", "Boolean", "__Schema",
+          "__Type", "__Field", "__InputValue", "__EnumValue", "__TypeKind",
+          "__Directive", "__DirectiveLocation" -> false
+          else -> true
+        }
+      }
+  )
+}
+
+/**
+ * Plain parsing, without validation or adding the builtin types
+ */
+private fun GQLDocument.Companion.parseInternal(inputStream: InputStream, filePath: String = "(source)"): GQLDocument {
```

Comment:
`= "(source)"` ?

Reply:
Yea, that's weird, if we parse from a String for an example, we cannot display the nice "preview". I guess I should just make filePath optional